app/models/digitization/newspaper.rb

```diff
@@ -0,0 +1,6 @@
+class Digitization::Newspaper < ApplicationRecord
+
+  validates :publication_code, uniqueness: { scope: [:year, :month, :day] }
+  validates :year, :month, :day, presence: true, if: :publication_code?
```

Comment:
I started going down the path of adding even more specific validation but not sure if it has added value
```suggestion
 validates :publication_code, uniqueness: { scope: [:year, :month, :day] }, format: { with: /[A-Z]{3}/, message: "%{value} should be three letters that represent the newspaper title" }
  validates :year, :month, :day, presence: true, if: :publication_code?
  validates :year, format: { with: /[0-9]{4}/, message: "%{value} should be a four digit year of publication"}
  validates :month, format: { with: /0[1-9]|1[0-2]/, message: "%{value} should be a two digit (zero padded) month"}
  validates :day, format: { with: /0[1-9]|[12]\d|3[01]/, message: "%{value} should be a two digit (zero padded) day of the month"}
```

See also https://ualbertalib-di.slack.com/archives/CSEHP0L0K/p1606408886290400

Reply:
My initial thought is that there isn't a ton of value because whatever the data in Peel is is right by definition, even if we can't construct a valid regex around it (if there's some one-off non-conforming issue or w/e)?